mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java

```diff
@@ -43,235 +45,266 @@
+            // the end of the file is reached, and we don't want those beeps to played one after the other as to:
+            // 1/ not lock the event thread
+            // 2/ have those beeps to end rather sooner than later
+            new Thread(() -> Toolkit.getDefaultToolkit().beep()).start();
+        }
+    }
+
+    public boolean isWrap() {
+        return textArea.getLineWrap();
+    }
+
+    ////////////////////////////
+    // Package-access methods //
+    ////////////////////////////
+
+    void wrap(boolean isWrap) {
+        textArea.setLineWrap(isWrap);
+        textArea.repaint();
+    }
+
+    void copy() {
+        textArea.copy();
+    }
+
+    void cut() {
+        textArea.cut();
+    }
+
+    void paste() {
+        textArea.paste();
+    }
+
+    void selectAll() {
+        textArea.selectAll();
+    }
+
+    void requestFocus() {
+        textArea.requestFocus();
+    }
+
+    JTextArea getTextArea() {
+        return textArea;
+    }
+
+    void addDocumentListener(DocumentListener documentListener) {
+        textArea.getDocument().addDocumentListener(documentListener);
+    }
+
+    void read(Reader reader) throws IOException {
+        // Feed the file's contents to text area
+        textArea.read(reader, null);
+
+        // If there are more than one lines, there is a line separator
+        lineSeparatorExists = textArea.getLineCount() > 1;
+
+        // Move cursor to the top
+        textArea.setCaretPosition(0);
+    }
+
+    void write(Writer writer) throws IOException {
+        Document document = textArea.getDocument();
+
+        // According to the documentation in DefaultEditorKit, the line separator is set to be as the system property
+        // if no other line separator exists in the file, but in practice it is not, so this is a workaround for it
+        if (!lineSeparatorExists)
+            document.putProperty(DefaultEditorKit.EndOfLineStringProperty, System.getProperty("line.separator"));
+
+        try {
+            textArea.getUI().getEditorKit(textArea).write(new BufferedWriter(writer), document, 0, document.getLength());
+        } catch (BadLocationException e) {
+            throw new IOException(e.getMessage());
+        }
+    }
+
+
+    public void setSyntaxBasedOnFilename(String fileExtension) {
+        // TODO use Apache Tika or some other way to work out mime-type (java built-in options suck)
+        String defaultMimeType = SyntaxConstants.SYNTAX_STYLE_NONE;
+        Map<String, String> extToMime = new HashMap<String, String>() {{
+            put("java", SyntaxConstants.SYNTAX_STYLE_JAVA);
+            put("js", SyntaxConstants.SYNTAX_STYLE_JAVASCRIPT);
+            put("gradle", SyntaxConstants.SYNTAX_STYLE_GROOVY);
+            put("c", SyntaxConstants.SYNTAX_STYLE_C);
+            put("cpp", SyntaxConstants.SYNTAX_STYLE_CPLUSPLUS);
+            put("html", SyntaxConstants.SYNTAX_STYLE_HTML);
+            put("htm", SyntaxConstants.SYNTAX_STYLE_HTML);
+        }};
```

Comment:
did you consider using FileTypeUtil#guessContentType instead?

Reply:
looks promising, will take a look.

Reply:
works good, PR updated